sql/sys_vars.cc

```diff
@@ -7655,6 +7656,25 @@ static Sys_var_ulonglong Sys_tf_sequence_table_max_upper_bound(
     GLOBAL_VAR(tf_sequence_table_max_upper_bound), CMD_LINE(REQUIRED_ARG),
     VALID_RANGE(1024, ULLONG_MAX), DEFAULT(1048576), BLOCK_SIZE(1));
 
+std::size_t buffered_error_log_size;
+
+static bool buffered_error_log_size_update(sys_var *, THD *, enum_var_type) {
+  buffered_error_log.resize(buffered_error_log_size);
+  return false;
+}
+
+static Sys_var_charptr Sys_var_buffered_error_log_filename(
+    "buffered_error_log_filename", "Filename of the buffered error log",
+    GLOBAL_VAR(buffered_error_log_filename), CMD_LINE(REQUIRED_ARG),
+    IN_FS_CHARSET, DEFAULT(""), NO_MUTEX_GUARD, NOT_IN_BINLOG, ON_CHECK(check_log_path));
+
+static Sys_var_ulonglong Sys_var_buffered_error_log_size(
+    "buffered_error_log_size", "Size of the buffered error log",
+    GLOBAL_VAR(buffered_error_log_size), CMD_LINE(REQUIRED_ARG),
+    VALID_RANGE(0, ULLONG_MAX), DEFAULT(0), BLOCK_SIZE(1), NO_MUTEX_GUARD,
+    NOT_IN_BINLOG, ON_CHECK(nullptr),
+    ON_UPDATE(buffered_error_log_size_update));
```

Comment:
I think we should not allow arbirtray sizes. what if we set just 100 bytes? and a single error message needs 200 bytes?
May be multiples of 16K or 1MB? or start value for which we saw decent improvement?

Reply:
Also, if the buffer is 1MB, the buffer is filled only until 512K, we will never see the remaining 512k until shutdown or crash?

Reply:
Yes, we only write on buffer full and on exit.

Reply:
@dutow can you please tell what happens when buffer size is 100 bytes and a single error message is 200 bytes?

Reply:
Since we are using an std::string as buffer and operator+= for adding the new log line to the buffer, it will automatically extend the buffer.